components/places/src/storage/bookmarks.rs

```diff
@@ -272,41 +278,53 @@ fn insert_bookmark_in_tx(db: &impl ConnExt, bm: &InsertableItem) -> Result<SyncG
     );
 
     let bookmark_type = bm.bookmark_type();
-    let params: Vec<(&str, &ToSql)> = match bm {
-        InsertableItem::Bookmark(ref b) => vec![
-            (":fk", &fk),
-            (":type", &bookmark_type),
-            (":parent", &parent.row_id),
-            (":position", &position),
-            (":title", &b.title),
-            (":dateAdded", &date_added),
-            (":lastModified", &last_modified),
-            (":guid", &guid),
-            (":syncStatus", &SyncStatus::New),
-            (":syncChangeCounter", &1),
-        ],
-        InsertableItem::Separator(ref _s) => vec![
-            (":type", &bookmark_type),
-            (":parent", &parent.row_id),
-            (":position", &position),
-            (":dateAdded", &date_added),
-            (":lastModified", &last_modified),
-            (":guid", &guid),
-            (":syncStatus", &SyncStatus::New),
-            (":syncChangeCounter", &1),
-        ],
-        InsertableItem::Folder(ref f) => vec![
-            (":type", &bookmark_type),
-            (":parent", &parent.row_id),
-            (":title", &f.title),
-            (":position", &position),
-            (":dateAdded", &date_added),
-            (":lastModified", &last_modified),
-            (":guid", &guid),
-            (":syncStatus", &SyncStatus::New),
-            (":syncChangeCounter", &1),
-        ],
+    // Annoyingly, we can't do (":title", &maybe_truncate_title(&item.title)),
+    // as the value won't live long enough. So we return it awkwardly.
```

Comment:
At this point, I think we could hoist the `db.execute_named_cached` call into each of the blocks, and pass a slice of params, avoiding the `Vec` allocation and title lifetime issue entirely.

Reply:
Good idea.